caput/config.py

```diff
@@ -397,6 +403,81 @@ def _prop(val):
     return prop
 
 
+def logging_config(default={}):
+    """
+    A Property type that validates the caput logging config.
+
+    Allows the type to be either a string (for backward compatibility) or a dict setting log levels
+    per module.
+
+    Parameters
+    ----------
+    default : optional
+        The optional default value.
+
+    Returns
+    -------
+    prop : Property
+        A property instance setup to validate the type.
+
+    Examples
+    --------
+    Should be used like::
+
+        class Project(object):
+
+            loglevels = logging_config({"root": "INFO", "annoying.module": "WARNING"})
+    """
+
+    def _prop(val):
+        config = val
+        if isinstance(config, str):
+            config = {"root": config}
+        elif not isinstance(config, dict):
+            raise ValueError(
+                "Expected a string or YAML block for config value 'logging', got "
+                "'{}'.".format(type(config.__name__))
+            )
+
+        # check entries, get module names and warn for duplicates when sorting into new dict
+        checked_config = {}
+        for key, level in config.items():
+            level = level.upper()
+            loglevels = ["DEBUG", "INFO", "WARNING", "ERROR", "NOTSET"]
```

Comment:
Python really should expose these publicly somewhere. Not much we can do about it though. I would move this declaration out of the loop though.

Reply:
you can pass a string to `setLevel` ...from python3.2